### ISSUE TITLE
Should name.givenName and name.familyName really be required?

### ISSUE THREAD

**Issue by s-andringa** [RIPAGlobal/scimitar#113]
Hi @pond,

We are running into problems with a client that sends us a `name` attribute with a blank `familyName` subattribute value. 

According to the [spec](https://datatracker.ietf.org/doc/html/rfc7643#section-4.1.1) neither `givenName` nor `familyName` are required subattributes, but in your schema [they are](https://github.com/RIPAGlobal/scimitar/blob/70492098ffb440de93c45b74cf83dffa7a48b94f/app/models/scimitar/schema/name.rb#L9-L10).

Now this client (Azure/MS Entra) is getting validation errors from our side, even though we don't strictly need those values.

I had to work around it by adding this to a Rails initializer:
```rb
# config/initializers/scimitar_patches.rb

Rails.application.configure do
  config.after_initialize do
    # Make familyName and givenName attributes optional as per SCIM 2.0 spec:
    Scimitar::Schema::Name.scim_attributes.each { |attribute| attribute.required = false }
  end
end
```

But I feel like these attributes should be optional by default. Is there a particular reason for making them required?

Cheers.

**Comment by pond** [comment]
Agreed. Inherited from originating gems I think.

This might be a breaking change for existing clients who *might* be relying on the Scimitar layer for validation but that seems like a distant edge case and unlikely; I don't think it warrants a major version bump nor does it warrant not fixing the requirement issue. If anyone hits trouble they can re-apply requirement via a similar initializer addition to the one you provide above.

With the test suite as-is, fixing the attributes to _not_ be required causes some failures. However, adding the following inside the `config.to_prepare` section of `config/initializers/scimitar.rb` of the dummy test app, for example:

```ruby
Scimitar::Schema::Name.scim_attributes.find { |a| a.name == 'familyName' }.required = true
Scimitar::Schema::Name.scim_attributes.find { |a| a.name == 'givenName'  }.required = true
```

...allows unmodified tests to pass, so that is reasonable proof that affected clients have an easy workaround.

Developing a PR now. Most of the time will be spent fixing up tests!

**Comment by pond** [comment]
Fix released in https://rubygems.org/gems/scimitar/versions/2.8.0. I'll backport to V1 within a week or two.

**Comment by pond** [comment]
Now available for Rails 6 via [v1.10.0](https://rubygems.org/gems/scimitar/versions/1.10.0) too.
